tests/integration/API/JSON/v1/SegmentsTest.php

```diff
@@ -162,6 +163,62 @@ public function testItReturnsErrorWhenTrashingSegmentWithActiveNewsletter() {
     expect($response->errors[0]['message'])->equals("List cannot be deleted because it’s used for 'Subject' email");
   }
 
+  public function testItReturnsErrorWhenTrashingSegmentWithActiveForm() {
+    $form = new FormEntity('My Form');
+    $form->setSettings([
+      'segments' => [(string)$this->segment3->getId()],
+    ]);
+    $this->entityManager->persist($form);
+    $this->entityManager->flush();
+
+    $response = $this->endpoint->trash(['id' => $this->segment3->getId()]);
+    $this->entityManager->refresh($this->segment3);
```

Comment:
I am also not familiar with Doctrine so take my answers with a grain of salt.
We call the method to reload the entity as we have made changes on the database.

Reply:
I think there is still something that I'm not fully understanding and I will investigate it a little bit more when I have the time, but we don't need to hold this PR because of that.

The test pass without this line (which is expected, after all, we are not checking for any of the values inside `$this->segment3`). But what intrigues me and makes me wonder that `refresh()` is not reloading `$this->segment3` instead is doing something else, is that if I add the following code below line 175, the test still passes and I would expect it to fail:

```
$this->assertNull($this->segment3->getDeletedAt());
```

Reply:
Thanks for the questions, they are helping me dive deeper into Doctrine. I will investigate too and let you know what I find.